provisioning/miniprov/hare_mp/main.py

```diff
@@ -243,11 +249,47 @@ def is_cluster_running() -> bool:
     return os.system('hctl status >/dev/null') == 0
 
 
+def nr_services() -> int:
+    cmd = ['hctl', 'status', '--json']
+    cluster_info = json.loads(execute(cmd))
+    # Don't include hax, count it just once later.
+    services = ['confd', 'ioservice', 's3server']
+    svcs_nr = 0
+    for node in cluster_info['nodes']:
+        for svc in node['svcs']:
+            if svc['name'] in services:
+                svcs_nr += 1
+    return (svcs_nr + 1)
+
+
+def all_services_started(url: str, nr_svcs: int) -> bool:
+    conf = ConfStoreProvider(url)
+    hostname = conf.get_hostname()
+    res = requests.get((f'http://127.0.0.1:8500/v1/kv/{hostname}/'
```

Comment:
I guess you can import hax.util and thus leverage ConsulUtil or ConsulKVBasic.
As you remember, this approach is quite fragile.

Reply:
I think we need to have retry mechanism while accessing consul here, ConsulUtil  suggested by Konstantin does that.

Reply:
Okay, let me try.

Reply:
build fails with,
```
hare_mp/main.py:41: error: Cannot find implementation or library stub for module named 'hax.util'     
hare_mp/main.py:41: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports 
Found 1 error in 1 file (checked 8 source files)                                                      
```

Reply:
Fixed.